tasks/MatchedCatalogsAnalysis.py

```diff
@@ -1,93 +1,42 @@
-import astropy.units as u
-import numpy as np
 import lsst.pipe.base as pipeBase
-from lsst.verify import Measurement
 import lsst.pex.config as pexConfig
-from sst_metrics_utils.filtermatches import filterMatches
-from lsst.verify.gen2tasks import register
-from lsst.verify.tasks import MetricTask, MetricConfig, MetricConnections, \
-    MetricComputationError
-from lsst.validate.drp.repeatability import calcPhotRepeat
+from lsst.verify.tasks import MetricTask, MetricConfig, MetricConnections
 
+from MeasureTasks import NumSourcesTask
 # The first thing to do is to define a Connections class. This will define all
 # the inputs and outputs that our task requires
 
 
-class NumSourcesTaskConnections(pipeBase.PipelineTaskConnections,
-                                dimensions=("tract", "patch", "abstract_filter",
-                                            "instrument", "skymap")):
+class MatchedCatalogAnalysisTaskConnections(MetricConnections,
+                                    dimensions=("tract", "patch", "abstract_filter",
+                                                "instrument", "skymap")):
     matchedCatalog = pipeBase.connectionTypes.Input(doc="Input matched catalog.",
                                                     dimensions=("tract", "patch", "instrument",
                                                                 "abstract_filter"),
                                                     storageClass="SimpleCatalog",
                                                     name="matchedCatalog")
-    measurements = pipeBase.connectionTypes.Output(doc="Number of sources in catalog.",
-                                                   dimensions=("tract", "patch", "instrument",
-                                                               "abstract_filter"),
-                                                   storageClass="MetricValue",
-                                                   name="nsrcMeas")
+    measurement = pipeBase.connectionTypes.Output(doc="Resulting matched catalog.",
+                                                  dimensions=("tract", "patch",
+                                                              "instrument","abstract_filter"),
+                                                  storageClass="MetricValue",
+                                                  name="metricvalue_{package}_{metric}")
 
 
-class NumSourcesTaskConfig(pipeBase.PipelineTaskConfig,
-                           pipelineConnections=NumSourcesTaskConnections):
-    pass
+class MatchedCatalogAnalysisTaskConfig(MetricConfig,
+                               defaultTemplates={"metric": None},
+                               pipelineConnections=MatchedCatalogAnalysisTaskConnections):
+    measure = pexConfig.ConfigurableField(
+        target=NumSourcesTask,  # This is the simplest measurement I can think of
```

Comment:
It seems that `NumSourcesTask` is acting as a placeholder here that is generally intended to be replaced (I think this is what the `python` option in the config yaml file does?). For readability, I wonder if it would be worth making this placeholder nature more clear. The current comment hints at this.

Reply:
I'll expand the comment to be clearer about what the default task is for.